doc/project/building.md

```diff
@@ -19,22 +19,52 @@ To produce a proper JavaScript file from your code, you need to call the linker:
+To make it do something, you need a main `js.JSApp` object:
+
+{% highlight scala %}
+import scala.scalajs.js
+
+object Main extends js.JSApp {
+  def main(): Unit = {
+    println("Hello world!")
+  }
+}
+{% endhighlight %}
+
+as well as the following sbt setting, which, to put it simply, turns your Scala.js project into an "application" rather than a "library":
+
+{% highlight scala %}
+scalaJSUseMainModuleInitializer := true
+{% endhighlight %}
+
+sbt will automatically detect the object that extends [`js.JSApp`]({{ site.production_url }}/api/scalajs-library/latest/#scala.scalajs.js.JSApp) (which must be unique), and use it as the main method of the application.
+Now, the .js file produced by `fastOptJS` will print `"Hello world!"`.
+
+Note that this will require that there is a *unique* object that extends `js.JSApp` or that the one to use be explicitly set with `mainClass in Compile := Some(<name>)`.
+If you explicitly set `mainClass`, note that it needs to be set on a per-configuration basis (i.e. the part `in Compile` is essential, otherwise the setting will be ignored). For further information see the Stack Overflow entry ['How to set mainClass in ScalaJS build.sbt?'](http://stackoverflow.com/questions/34965072/how-to-set-mainclass-in-scalajs-build-sbt) (specific to Scala.js) and the Stack Overflow entry ['How to set main class in build?'](http://stackoverflow.com/questions/6467423/how-to-set-main-class-in-build) (not specific to Scala.js).
+
 ## Running in the console
 
-You can run your application by using the `run` task:
+You can run a Scala.js application (that has `scalaJSUseMainModuleInitializer` set to `true`) by using the `run` task:
 
     sbt> run
 
-This will detect and run classes that extend
-[`js.JSApp`]({{ site.production_url }}/api/scalajs-library/{{ site.versions.scalaJS }}/#scala.scalajs.js.JSApp), while optionally prompting the user to choose a class if multiple such classes exist (fails with multiple classes if `persistLauncher := true`, see section below for details).
-
-By default, this command uses [Node.js](http://nodejs.org/), which you need to install separately, to run the produced JavaScript code.
+This will run the `-fastopt.js` file right inside of your sbt console.
+By default, the file is run with [Node.js](http://nodejs.org/), which you need to install separately.
 If your application or one of its libraries requires a DOM (which can be specified with `jsDependencies += RuntimeDOM`), you will also need to install [`jsdom`](https://github.com/tmpvar/jsdom) with `npm install jsdom`.
 
 There are alternative JavaScript interpreters that are available.
 See [JavaScript environments](./js-environments.html) for more details.
 
-### Disabling the optimizations
+### Deprecated: Run without `scalaJSUseMainModuleInitializer`
+
+It is still possible to `run` a Scala.js application that does not have `scalaJSUseMainModuleInitializer := true`.
+However, this is not recommended anymore.
```

Comment:
I'm not sure it's actually useful.

I'm not certain we'll not have to reinstate that feature in some way, but for now I figured it doesn't hurt to tell people they shouldn't use it. If we don't, we'll never get any complaints, and we won't know that there are use cases for the feature.